http/RemoteResource.cc

```diff
@@ -421,7 +344,119 @@ namespace http {
+        }
+
+        //  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -
+        // If we are filtering the response (for example to inject data URL into a dmr++ file),
+        // The file is locked and we have the information required to make the substitution.
+        // This is controlled by:
+        //  - The template_key string must not be empty.
+        if(!template_key.empty()){
+            unsigned int count = filter_retrieved_resource(template_key, replace_value);
+            BESDEBUG(MODULE, prolog << "Replaced " << count <<
+                                    " instance(s) of template(" <<
+                                    template_key << ") with " << replace_value << " in cached RemoteResource" << endl);
+        }
+
+        //  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -
+        // I think right here is where I would be able to cache the data type/response headers. While I have
+        // the exclusive lock I could open another cache file for metadata and write to it.
+        {
+            // FIXME THIS IS WHERE WE NEED TO USE NGAP HASH CACHE FILE NAMES NOT FLC NAMES.
+            string hdr_filename = d_resourceCacheFileName + ".hdrs";
+            std::ofstream hdr_out(hdr_filename.c_str());
+            try {
+                for (size_t i = 0; i < this->d_response_headers->size(); i++) {
+                    hdr_out << (*d_response_headers)[i] << endl;
+                }
+            }
+            catch (...) {
+                // If this fails for any reason we:
+                hdr_out.close(); // Close the stream
+                unlink(hdr_filename.c_str()); // unlink the file
+                unlink(d_resourceCacheFileName.c_str()); // unlink the primary cache file.
+                throw;
+            }
+        }
+        // #########################################################################################################
+
+        // Change the exclusive lock on the new file to a shared lock. This keeps
+        // other processes from purging the new file and ensures that the reading
+        // process can use it.
+        cache->exclusive_to_shared_lock(d_fd);
+        BESDEBUG(MODULE, prolog << "Converted exclusive cache lock to shared lock." << endl);
+
+        // Now update the total cache size info and purge if needed. The new file's
+        // name is passed into the purge method because this process cannot detect its
+        // own lock on the file.
+        unsigned long long size = cache->update_cache_info(d_resourceCacheFileName);
+        BESDEBUG(MODULE, prolog << "Updated cache info" << endl);
+
+        if (cache->cache_too_big(size)) {
+            cache->update_and_purge(d_resourceCacheFileName);
+            BESDEBUG(MODULE, prolog << "Updated and purged cache." << endl);
+        }
+        BESDEBUG(MODULE, prolog << "END" << endl);
+        d_initialized = true;
```

Comment:
Just reading the headers is not sufficient to label the instance as initialized.
Setting the state variable `d_initialized` should happen in the calling methods that orchestrate the entire initialization/inflation of the object.

Reply:
so move the "d_initialized = true" to after the method call

Reply:
should the same be done with the load_hdrs_from_files()?